src/utils/dbConnection.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+
+const dbConnection = (function () {
+	const url = process.env.MONGO_DB_URL || 'mongodb://localhost:27017';
+	let mongoClient; 
+
+	async function dbConnect() {
+		mongoClient = await mongoose.connect(url, {
+			useNewUrlParser: true,
+			poolSize: 5,
+		});
+		return mongoClient;
+	}
+
+	async function getdbClient() {
+		try {
+			if (mongoClient) {
```

Comment:
if connection get stale then also the variable is not null and we will not be forming the new connection